polars/polars-lazy/polars-plan/src/dsl/function_expr/schema.rs

```diff
@@ -59,7 +59,38 @@ impl FunctionExpr {
                     }
                     #[cfg(feature = "timezones")]
                     TzLocalize(tz) => return mapper.map_datetime_dtype_timezone(Some(tz)),
-                    DateRange { .. } => return mapper.map_to_supertype(),
+                    DateRange {
+                        every: _,
+                        closed: _,
+                        tz,
+                    } => {
+                        let mut ret = mapper.map_to_supertype()?;
```

Comment:
Can we hoist this into a function. That keeps the `schema` branch a bit more lean.